qiskit_nature/problems/sampling/protein_folding/__init__.py

```diff
@@ -124,6 +124,20 @@
 from .peptide.chains.side_chain import SideChain
 from .peptide.peptide import Peptide
 from .protein_folding_problem import ProteinFoldingProblem
+from ....deprecation import warn_deprecated, DeprecatedType, NatureDeprecationWarning
+
+warn_deprecated(
+    "0.5.0",
+    old_type=DeprecatedType.PACKAGE,
+    old_name="qiskit_nature.problems.sampling.protein_folding",
```

Comment:
Do we want to deprecate protein folding only or the full sampling package?

Reply:
I was wondering the same. Technically, only protein folding is being moved to that location in Qiskit Research. Afterwards, `sampling` is empty, so I am not sure whether we still need to deprecate this specific (now empty) module explicitly..

@woodsp-ibm any thoughts? Once we have a migration guide which outlines all the code being moved to `second_q`, this could be mentioned there, too

Reply:
In principle, someone could have been implementing their own algorithm based on the `SamplingProblem` interface, in their fork (which is outside `protein_folding` but inside `sampling`. They might sync with our `main` from time to time. So it would be reasonable to show deprecation warnings regarding `sampling` too, I think (but the impact is minor). 

I'm assuming that the whole `sampling` package will disappear from Nature, as we are narrowing the scope.

Reply:
Right, that's a good point! @manoelmarques could you please also add a separate deprecation warning to the `sampling` module (or the `SamplingProblem` if it cannot go on the module itself) that simply states, this is being removed in order to focus the scope of Qiskit Nature?

Reply:
Seems reasonable that the whole `sampling` module is deprecated and eventually removed from here - with the code going elsewhere and being maintained there. `sampling` is not under `second_q` so does not remain here at all. Had someone used SamplingProblem as a base then they will need to  change code to use the version in the new location. The migration guide from 0.4.0 to 0.5.0 will need to include sampling/Protein folding, where it is now, and to migrate over to it etc.